File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1

        while l <= r:
            mid = (l + r) // 2
            print(l, r, mid)
            if target == nums[mid]:
                return mid
            elif target < nums[mid]:
                r = mid - 1
            else:
                l = mid + 1

        return -1

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-1.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, greater<int>> pq;

        for(auto elem: nums){
            if(k == 0){
                if(pq.top() < elem){
                    pq.pop();
                    pq.push(elem);
                }
            }
            else{
                pq.push(elem);
            }
            if(k) k--;
        }
        return pq.top();
    }
};
